tests/test_bot.py

```diff
@@ -2446,6 +2447,61 @@ async def test_get_updates(self, bot):
         if updates:
             assert isinstance(updates[0], Update)
 
+    @pytest.mark.parametrize("bot_class", [Bot, ExtBot])
+    async def test_get_updates_read_timeout_deprecation_warning(
+        self, bot, recwarn, monkeypatch, bot_class
+    ):
+        # Using the normal HTTPXRequest should not issue any warnings
+        await bot.get_updates()
+        assert len(recwarn) == 0
+
+        read_timeout = None
+
+        async def catch_timeouts(*args, **kwargs):
+            nonlocal read_timeout
+            read_timeout = kwargs.get("read_timeout")
+            return HTTPStatus.OK, b'{"ok": "True", "result": {}}'
+
+        monkeypatch.setattr(HTTPXRequest, "read_timeout", BaseRequest.read_timeout)
+        monkeypatch.setattr(HTTPXRequest, "do_request", catch_timeouts)
+
+        bot = bot_class(get_updates_request=HTTPXRequest(), token=bot.token)
+        await bot.get_updates()
+
+        assert len(recwarn) == 1
+        assert "does not override the property `read_timeout`" in str(recwarn[0].message)
+        assert recwarn[0].category is PTBDeprecationWarning
+        assert recwarn[0].filename == __file__, "wrong stacklevel"
+
+        assert read_timeout == 2
+
+    @pytest.mark.parametrize(
+        ("read_timeout", "timeout", "expected"),
+        [
+            (None, None, 0),
+            (1, None, 1),
+            (None, 1, 1),
+            (DEFAULT_NONE, None, 10),
+            (DEFAULT_NONE, 1, 11),
+            (1, 2, 3),
+        ],
+    )
+    async def test_get_updates_read_timeout_value_passing(
```

Comment:
we should do the same test but with applicationbuilder instead of Bot()? Just to confirm if the right values are being passed.

Reply:
yeah, won't hurt :)